source/common/network/dns_impl.cc

```diff
@@ -67,46 +67,69 @@ void DnsResolverImpl::initializeChannel(ares_options* options, int optmask) {
   ares_init_options(&channel_, options, optmask | ARES_OPT_SOCK_STATE_CB);
 }
 
-void DnsResolverImpl::PendingResolution::onAresHostCallback(int status, int timeouts,
-                                                            hostent* hostent) {
+void DnsResolverImpl::PendingResolution::onAresGetAddrInfoCallback(int status, int timeouts,
+                                                                   ares_addrinfo* addrinfo) {
   // We receive ARES_EDESTRUCTION when destructing with pending queries.
   if (status == ARES_EDESTRUCTION) {
     ASSERT(owned_);
     delete this;
+
+    if (addrinfo) {
```

Comment:
Can this actually happen? Seems unlikely?

Reply:
Yes. I'll remove it.